lib/PHPExif/Adapter/ImageMagick.php

```diff
@@ -46,7 +46,15 @@ public function getExifFromFile(string $file) : Exif
         $data_filename = basename($file);
         $data_filesize = filesize($file);
         $mimeType = mime_content_type($file);
-        $additional_data = array('MimeType' => $mimeType, 'filesize' => $data_filesize, 'filename' => $data_filename);
+        $data_width = $im->getImageWidth();
+        $data_height = $im->getImageHeight();
+        $additional_data = array(
+            'MimeType' => $mimeType,
+            'filesize' => $data_filesize,
+            'filename' => $data_filename,
+            'width' => $data_width,
+            'height' => $data_height);
```

Comment:
If we are on it, we may as well replace the legacy `array()`-function be the more efficient language construct `[]`.

```suggestion
        $additional_data = [
            'MimeType' => $mimeType,
            'filesize' => $data_filesize,
            'filename' => $data_filename,
            'width' => $data_width,
            'height' => $data_height
        ];
```

But my actual question out of curiosity is another one: Why do we need to add `width` and `height` here if it hasn't been necessary before?

Reply:
It goes back to the comment I made on Gitter: while this code is not new, it was only used for raw files in the past. It is now (after the merge of https://github.com/LycheeOrg/Lychee/pull/1309) used for JPEGs as well if exiftool is missing, which is a lot more common usecase. So I'm guessing that the bug was "always" there, it just wasn't noticed until now due to the relative obscurity of this code...

Reply:
Understood, but I wonder whether this is only a temporary problem and wether we actually should go the opposite path and remove these additional attributes entirely and consistently. Of course, this would be a strategic change. It would also be fine for me to make theses changes as contained in this PR now, because it is a quick fix and then remove them later again, if we decide for it.

Let me elaborate on what I mean.

These additional attributes we are considering here (i.e. file size, MIME type, fie name) are no true part of the EXIF information. I assume that ExifTool outputs them (together with a lot of other non-EXIF data like inode number, file permissions, etc.) and one originally decided to add these data for the other EXIF readers as well for consistency reasons. But actually, we are extremely inconsistent about them when it comes to spelling (i.e. capitalization, hyphenation.)

Here (in `Adapter/ImageMagick`) we use `MimeType`, `filesize` and `filename`. In  `Adapter/Native` we use camel-case everywhere as in `MimeType`, `FileSize` and `FileName`. And in `Adapter/FFProbe` we use a different variant again. That is actually the reason why am convinced that unstructured arrays to exchange data are a really bad thing.

Adding these additional attributes as part of the EXIF extraction requires extra file system calls. In order to extract the image height and with we actually load the image. The Imagick adapter loads the image via Imagick (naturally) and the native adapter uses GD image operations, i.e. `getimage`. This is something which we do anyway in Lychee to auto-rotate the image and generate the smaller image variants. As the file size, width and height may have changed after that, we re-read these attributes and do not rely on the values provided by this library.

When I added the myriad a tests as part of my PR #1351, I also added non-typical image files (TIFF, WEBP and raw examples) and also let the whole test bench run without Imagick. See here

 - [`PhotosAddHandlerTestAbstract`](https://github.com/LycheeOrg/Lychee/blob/use_filestreams/tests/Feature/PhotosAddHandlerTestAbstract.php); instantiated by
    - [`PhotosAddHandlerImagickTest`](https://github.com/LycheeOrg/Lychee/blob/use_filestreams/tests/Feature/PhotosAddHandlerImagickTest.php)
    - [`PhotosAddHandlerGDTest`](https://github.com/LycheeOrg/Lychee/blob/use_filestreams/tests/Feature/PhotosAddHandlerGDTest.php)
 - [`PhotosRotateTestAbstract`](https://github.com/LycheeOrg/Lychee/blob/use_filestreams/tests/Feature/PhotosRotateTestAbstract.php); instantiated by
    - [`PhotosRotateImagickTest`](https://github.com/LycheeOrg/Lychee/blob/use_filestreams/tests/Feature/PhotosRotateImagickTest.php)
    - [`PhotosRotateGDTest`](https://github.com/LycheeOrg/Lychee/blob/use_filestreams/tests/Feature/PhotosRotateGDTest.php)

(BTW: This PR is still pending for review, no pressure :smile:) This increased test coverage to 92% :tada: and unveiled a lot, I mean a LOT, of lurking errors which were unrelated to the actual PR. But hopefully, these tests will prevent regressions bugs like #1367 from happening again in the future, because every combination of photo type and raw image with an EXIF handler is now tested. (Embrace yourself for much longer test run-times than before). However, I fixed these problems on the Lychee side. For example, Lychee does not rely on these attributes as part of the EXIF data anymore.

In summary, I would suggest to do the following:

 - We add these additional attributes as part of this PR. This provide us with a quick fix for #1367 on the current master. I guess PR #1351 still requires some time to get merged.
 - In the long run I would rather prefer
    - to remove this additional non-truly EXIF attributes from the EXIF library, i.e. instead of adding them because EXIFtool does, I would remove them everywhere
    - replacing this unfortunate, non type-safe array `$data` by something more reliable